include/bitcoin/bitcoin/utility/pseudo_random.hpp

```diff
@@ -0,0 +1,125 @@
+/**
+ * Copyright (c) 2011-2017 libbitcoin developers (see AUTHORS)
+ *
+ * This file is part of libbitcoin.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+#ifndef LIBBITCOIN_PSEUDO_RANDOM_HPP
+#define LIBBITCOIN_PSEUDO_RANDOM_HPP
+
+#include <random>
+#include <cstdint>
+#include <bitcoin/bitcoin/constants.hpp>
+#include <bitcoin/bitcoin/define.hpp>
+#include <bitcoin/bitcoin/utility/asio.hpp>
+#include <bitcoin/bitcoin/utility/data.hpp>
+
+namespace libbitcoin {
+
+/**
+ * DEPRECATED
+ * Generate a pseudo random number within the domain.
+ * @return  The 64 bit number.
+ */
+BC_API uint64_t pseudo_random();
+
+/**
+ * DEPRECATED
+ * Generate a pseudo random number within [begin, end].
+ * @return  The 64 bit number.
+ */
+BC_API uint64_t pseudo_random(uint64_t begin, uint64_t end);
+
+/**
+ * DEPRECATED
+ * Fill a buffer with randomness using the default random engine.
+ */
+BC_API void pseudo_random_fill(data_chunk& out);
+
+/**
+ * DEPRECATED
+ * Convert a time duration to a value in the range [max/ratio, max].
+ * @param[in]  maximum  The maximum value to return.
+ * @param[in]  ratio    The determinant of the minimum duration as the inverse
+ *                      portion of the maximum duration.
+ * @return              The randomized duration.
+ */
+BC_API asio::duration pseudo_randomize(const asio::duration& maximum,
+    uint8_t ratio=2);
+
+
+class BC_API pseudo_random
+{
+  public:
+    /**
+     * Fill a container with randomness using the default random engine.
+     */
+    template<class Container>
+    static void fill(Container& out)
```

Comment:
This will work for now, but we eventually hit problems with header inclusion of template methods.

Reply:
Where?  Just wondering.  I have been porting the -network changes for the other PR with each revision of this and haven't hit them yet.